### PR TITLE
test: add tests for azure and GCP CSFLE, fix aws boto3 error

### PR DESCRIPTION
**Pull request by emadum**
Ports #2662 and #2720 for 4.0

NODE-2825
NODE-3056

### CODE REVIEW

File: test/functional/client_side_encryption/prose.test.js [hunk 1]
```diff
@@ -4,6 +4,18 @@ const chai = require('chai');
 const expect = chai.expect;
 chai.use(require('chai-subset'));
 
+const EJSON = require('mongodb-extjson');
```
Comment (nbbeeken) [inline test/functional/client_side_encryption/prose.test.js:7]: Can we use EJSON from bson here? and in doing so avoid bringing in the deprecated `mongodb-extjson` lib
`const { EJSON } = require('bson')`

File: .evergreen/run-tests.sh [hunk 2]
```diff
@@ -53,7 +53,7 @@ if [[ -z "${CLIENT_ENCRYPTION}" ]]; then
   unset AWS_ACCESS_KEY_ID;
   unset AWS_SECRET_ACCESS_KEY;
 else
-  npm install mongodb-client-encryption
+  npm install mongodb-client-encryption@1.1.1-beta.0
```
Comment (nbbeeken) [inline .evergreen/run-tests.sh:56]: Is this to ensure we get in the (semi) recent changes to add GCP and Azure? I think we'd be better off using a range so we can get the latest as we make more releases: `npm install mongodb-client-encryption@">=1.2.1"`

**Review by nbbeeken — CHANGES_REQUESTED** [review]
Testing LGTM, just some CI notes

File: .evergreen/run-tests.sh (hunk 2 above)
Comment (emadum) [reply .evergreen/run-tests.sh:56]: That change came in from the 3.6 PR (#2662) so I'm not entirely sure, but agreed using a range is better.

Comment (nbbeeken) [reply .evergreen/run-tests.sh:56]: Ah my mistake, I missed that came from there we should correct it there too.

**Review by nbbeeken — APPROVED** [review]
LGTM

**Review by durran — APPROVED** [review]

**Pull request merged**